### PR TITLE
[net processing] Move Misbehaving() to PeerManager

### PR DESCRIPTION
**Pull request by jnewbery**
Continues the work of moving net_processing logic into PeerLogicValidation. See https://github.com/bitcoin/bitcoin/pull/19704 and https://github.com/bitcoin/bitcoin/pull/19607#discussion_r462032894 for motivation.

This PR also renames `PeerLogicValidation` to `PeerManager` as suggested in https://github.com/bitcoin/bitcoin/pull/10756#pullrequestreview-53892618.

### CODE REVIEW

**Comment by theStack** [comment]
The PR title only addresses one of the many commits (the last one), I assume you either wanted to only include this commit or change the title to be more general to also include all other changes.

**Comment by jnewbery** [comment]
> The PR title only addresses one of the many commits (the last one), I assume you either wanted to only include this commit or change the title to be more general to also include all other changes.

The object of this PR is to move Misbehaving to PeerLogicValidation. The other commits are necessary to do that because they all eventually call in to Misbehaving: https://doxygen.bitcoincore.org/net__processing_8cpp.html#ad2983e754c4d90bd68adf91b208664f5

**Comment by theStack** [comment]
> > The PR title only addresses one of the many commits (the last one), I assume you either wanted to only include this commit or change the title to be more general to also include all other changes.
> 
> The object of this PR is to move Misbehaving to PeerLogicValidation. The other commits are necessary to do that because they all eventually call in to Misbehaving: https://doxygen.bitcoincore.org/net__processing_8cpp.html#ad2983e754c4d90bd68adf91b208664f5

Thanks for clarifying, that makes sense. (And it's the first time in my life that a generated caller graph by Doxygen actually is of any help and not confusing ;-))
Concept ACK

**Comment by MarcoFalke** [comment]
Concept ACK

**Comment by MarcoFalke** [comment]
( Note to myself: commit eb170099f3ea91555b56fdaaae89a64fe04b93be with this diff https://github.com/bitcoin/bitcoin/pull/19704#discussion_r474475437 should pass the unit tests again )

**Comment by jnewbery** [comment]
> ( Note to myself: commit eb17009 with this diff #19704 (comment) should pass the unit tests again )

Thanks @MarcoFalke . I've cherrypicked that commit into this branch, which reduced further the number of parameters in some of these moved functions.

**Comment by jnewbery** [comment]
Rebased

**Comment by practicalswift** [comment]
Concept ACK

**Comment by MarcoFalke** [comment]
Almost ACK. Please use a scripted diff where appropriate:

```
 7:  534b4a2d86 !  7:  fffffffffff [net processing] Rename PeerLogicValidation to PeerManager
    @@ Metadata
     Author: John Newbery <john@johnnewbery.com>
     
      ## Commit message ##
    -    [net processing] Rename PeerLogicValidation to PeerManager
    +    scripted-diff: [net processing] Rename PeerLogicValidation to PeerManager
    +
    +    -BEGIN VERIFY SCRIPT-
    +    sed -i 's/PeerLogicValidation/PeerManager/g' $(git grep -l PeerLogicValidation ./src ./test)
    +    -END VERIFY SCRIPT-
     
         PeerLogicValidation was originally net_processing's implementation to
         the validation interface. It has since grown to contain much of
```


ACK b9486cc514 🍳

Exclusively move-only and rename-only changes. The only risk I see would be
uninteded changes due to shadowing, but that shouldn't be a problem because all
local symbols are aliases to the global ones anyway. (m_mempool == ::mempool,
m_params == Params(), ...)

* d45f2bb682 --color-moved=dimmed-zebra --color-moved-ws=ignore-all-space
* e4f6ec6aa3 --word-diff-regex=. -U0
* 534b4a2d86 I have not reviewed this. Why is this not a scripted diff ???
* 66f20e7738 --word-diff-regex=. -U0
* f9a76b620b --color-moved=dimmed-zebra --color-moved-ws=ignore-all-space 
* e83b1497a7 --word-diff-regex=. -U0
* 689b2b3926 --color-moved=dimmed-zebra --color-moved-ws=ignore-all-space
* f789dd7214 --word-diff-regex=. -U0
* b9486cc514 --word-diff-regex=. -U0 (also reviewed with b9486cc514 --color-moved=dimmed-zebra --color-moved-ws=ignore-all-space)

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK b9486cc514 🍳

Exclusively move-only and rename-only changes. The only risk I see would be
uninteded changes due to shadowing, but that shouldn't be a problem because all
local symbols are aliases to the global ones anyway. (m_mempool == ::mempool,
m_params == Params(), ...)

* d45f2bb682 --color-moved=dimmed-zebra --color-moved-ws=ignore-all-space
* e4f6ec6aa3 --word-diff-regex=. -U0
* 534b4a2d86 I have not reviewed this. Why is this not a scripted diff ???
* 66f20e7738 --word-diff-regex=. -U0
* f9a76b620b --color-moved=dimmed-zebra --color-moved-ws=ignore-all-space 
* e83b1497a7 --word-diff-regex=. -U0
* 689b2b3926 --color-moved=dimmed-zebra --color-moved-ws=ignore-all-space
* f789dd7214 --word-diff-regex=. -U0
* b9486cc514 --word-diff-regex=. -U0 (also reviewed with b9486cc514 --color-moved=dimmed-zebra --color-moved-ws=ignore-all-space)
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUi44wv9EqnP0JBiaYaFXQOP16rPsjbXFLjfNTDcxy44OW6fdjz+y3twK6r5dSvj
QIsy5XuOL9Wb+RONSaBIfuABsKHZPxP3gPP6PrWd/ukD4ypUyvbi97UrCYGgaGkb
A0H6n4OIF2o9QXo+pu/fLFfPucuy/CWML5BrhUAmVwjjB7WLs5wFfGyX7nleL9nd
pM4R/JK9wgttCScaVBhF+hvWJazV86UqtktLU++AR675mH4YjRBx2Ng3BQRrmeD5
bScdeICbZ9uKYKDjZiyegfwmr7ni+zP7vIc1LVNbCEXBYzfMSyeB3+fKwQl0+bRt
ihgTP9FhuMcfGt6NLTAr5/J/Ar4IFlFYWceB7n/DG38Tt0cK1cs+9yiTyhRXttsK
ZL5u1oYqA9OMf6tLm1Ey0MvNgYRJ7iQMj1xM4BWZJlztUdaOseOT4tTaY9yQYuQg
lZJwmTGfUXodw65qfU07ARGigKsCQzL9qchrzHlyc+oQHSAcwrSTMmFKlPxit53W
pmMky4Ak
=kiZz
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `5be0c45ea49e947755a8bba146134d0a71a0de60deb5f538b4a22c3d9a457df9  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401085be0c45ea49e947755a8bba146134d0a71a0de60deb5f538b4a22c3d9a457df9f01047e56fcc3d895c174608970ade88cf9b08fff01036fe8f14c1ffa7d94177f8db39e7dc0108f1045f49fe7ef008f9b4001b38475e470083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff0105cf77656d57eeb4077a5fd157779346e08f12083a626b394441c93dde83d5176d9608ccf592d277c974a92ea2891ecb1bcae3908f1045f49fe7ef0086134da78201313940083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff01076f39b1e93d24f67fd487340938a673008f1207b0956742da725d3351bb9794fbb4967beca4e0a8cfd0afa7aa24355c44b4f8908f020d5116060b0cdc49a58e591eb909048acb3b9b7ccdf24fa628c3ecf9ed2c89e0308f1045f49fe7ff00888f592dda9f774db0083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010ed36c818305279a77d38a5842aa6616c08f120052676d3f234f5c8c101a3f32fb38b92865ac76364202b3daea6285f9308bc8908f0205ece7f67a5ba59312c6278295fbf70fcdf18f9fc557021f92703aa6c96375f4f08f1045f49fe7ef008477488d14b8832cb0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

**Comment by jnewbery** [comment]
Thanks for the review @MarcoFalke ! I've taken your suggestion of making the rename a scripted diff (and added a follow-up commit that fixes the indentation).

**Comment by MarcoFalke** [comment]
re-ACK ab9e416812 🔋

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK ab9e416812 🔋
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgOmAv/XiLCgaJ+N1uXLqQGle28I8T65NwC7lN8op1+AcF0HdvG4P/nr9XAKJGO
HGutxFBbwYhY0gdFiShEKPI/LdTIPF+HEARqQ3bt4HCM1XxFf01UZOKIRjDpEOIl
Zij/8/JIbYQnrl5ROhLlDc89QQKxOyfcviYkshahiXK3QUEDbLrb8axuWl4OJ7jR
eXvlwHqY54k8DbT8ofjCxaHpADgWjLWxsjW4a3QhYM0MnI+KI8g0UryLBvGs2A23
X6VTgJ0hXQX+QJTLVUjy0jwIalrBVWpMXRcDa+LF6PTsZnFKIH3i9ej2p+91MHJk
2kxiAUFYcbpYIhhAi21m4TUSwZ2OjxXPv1Fa3X122caj2eHmotlrH+ACo6/O0uWe
3QWyAB5caNLs0KAxtCC5F7A2caYKjlVw/BnR8krn8PQXP+H3kjZAjUzDQkT9xOhx
eHAVHR5w7+dDbuZQ39fdC9inTJqXSYzwtvfMPpxpCIPECleG0SxvrtRTw5UWzOnU
OUb2Ix1f
=ffIE
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `8ca020057d8fe8423b0403eeebc5772bea83a87127a66a83c6409a6a47884972  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401088ca020057d8fe8423b0403eeebc5772bea83a87127a66a83c6409a6a47884972f010e8df1f24178940cebcac14eede89ae3e08fff0101713b9c71ea5d8b86508484146ee730808f1045f4a2afff008d23d51214255d7190083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0105618059da74a5d377ebef40abcb06bff08f1045f4a2afff008b145325aa63a5fec0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0105afd4606302283d2db16f1296478590e08f1045f4a2afff008e0bcc51bb813d9090083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df01060dd60e7dd9a14641e4f52f1606e00c208f1045f4a2afff008447f22ed24bdd8200083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Comment by hebasto** [comment]
Concept ACK.

**Review by hebasto — COMMENTED** [review]
IIUC, the goal of this PR is a better encapsulation. Therefore, functions that do not require an access to private data members should be placed into a dedicated namespace as non-member ones. I mean `MaybePunishNodeForBlock()`, `MaybePunishNodeForTx()` and `Misbehaving()`.

In commit 30dbd283cc3c4bcc55cefe5925581dbf9da78c2d "scripted-diff: [net processing] Rename PeerLogicValidation to PeerManager" the commit message is incomplete.

**Comment by jnewbery** [comment]
Please see https://github.com/bitcoin/bitcoin/pull/19607#discussion_r462032894 for motivation. The idea is to move the `g_peer_map` to be a member of `PeerLogicValidation`. To do that, we need to move these functions into `PeerLogicValidation` since they need to access that data.

File: src/net_processing.h [hunk 1]
```diff
@@ -105,6 +105,7 @@ class PeerManager final : public CValidationInterface, public NetEventsInterface
      */
     bool MaybeDiscourageAndDisconnect(CNode& pnode);
 
+    void ProcessOrphanTx(std::set<uint256>& orphan_work_set, std::list<CTransactionRef>& removed_txn) EXCLUSIVE_LOCKS_REQUIRED(cs_main, g_cs_orphans);
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply src/net_processing.h:124]: Done!

**Comment by hebasto** [comment]
> In commit [30dbd28](https://github.com/bitcoin/bitcoin/commit/30dbd283cc3c4bcc55cefe5925581dbf9da78c2d) "scripted-diff: [net processing] Rename PeerLogicValidation to PeerManager" the commit message is incomplete.

I meant that before the latest push the commit message was
```
scripted-diff: [net processing] Rename PeerLogicValidation to PeerManager

-BEGIN VERIFY SCRIPT-
sed -i 's/PeerLogicValidation/PeerManager/g' $(git grep -l PeerLogicValidation ./src ./test)
-END VERIFY SCRIPT-

PeerLogicValidation was originally net_processing's implementation to
the validation interface. It has since grown to contain much of
```

I didn't mean to drop the scripted-diff part. Sorry.

**Comment by hebasto** [comment]
> The idea is to move the `g_peer_map` to be a member of `PeerLogicValidation`. To do that, we need to move these functions into `PeerLogicValidation` since they need to access that data.

But `MaybePunishNodeForBlock()`, `MaybePunishNodeForTx()` and `Misbehaving()` do not access to `g_peer_map`, no?

**Comment by MarcoFalke** [comment]
> But MaybePunishNodeForBlock(), MaybePunishNodeForTx() and Misbehaving() do not access to g_peer_map, no?

Misbehaving calls `GetPeerRef`

**Comment by hebasto** [comment]
> > But MaybePunishNodeForBlock(), MaybePunishNodeForTx() and Misbehaving() do not access to g_peer_map, no?
> 
> Misbehaving calls `GetPeerRef`

Correct. But it seems that `MaybePunishNodeForBlock()` and `MaybePunishNodeForTx()` have neither direct nor indirect access to `g_peer_map`.

**Comment by MarcoFalke** [comment]
They call `Misbehaving`, which calls `GetPeerRef`, which accesses `g_peer_map`

**Comment by hebasto** [comment]
> They call `Misbehaving`, which calls `GetPeerRef`, which accesses `g_peer_map`

Indeed. Sorry for noise.

**Comment by jnewbery** [comment]
> I didn't mean to drop the scripted-diff part. Sorry.

Oops. I didn't mean to do that. Restored the scripted-diff and fixed the commit log.

> > They call Misbehaving, which calls GetPeerRef, which accesses g_peer_map

> Indeed. Sorry for noise.

No problem. Thanks for reviewing!

**Review by hebasto — APPROVED** [review]
ACK 7ed1612715785bb48053c50167c136f1d46ac352

**Comment by MarcoFalke** [comment]
re-ACK 7ed1612715 only change is in commit message 💮

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK 7ed1612715 only change is in commit message 💮
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUjjSAv/eL83MPVWHHsG87sMVTjfZH87F5ioMxNs/Rj/ZwuFkj3yMLwXFPDlJMsw
cBoTZS/i1PBCZwueOjGJd5E5RFY98FsLsUG6zsxcnxZvbiUCnGvc6gnPqHnFLDzN
yJGDOLlFrT9NRMsuo+sQ9fZ0dGRB3UmXkX9g3PRIFqLAvW9++ZHgHsHFWEDaWjXo
eZ40OcEAZ0/y8OiZLNcyP5m/XUrmirxLQvLBbnBa7knP9ieE8Gz1XIgTPXrbJxno
PzPIwFrB0z6iyKB1v8HW1mO/qK+zGTKkPAjev0KucuFmlApTp+WcH98yBvhazuQQ
GMBk64LQLOhb2RiHTrjKHUXfcr05gc68or4LbB40E/8FoSHWu5H3U6h3rUgOWp5+
xklUznWZix1ajWjIJxDBDALKgxJNkDQrtRTVX0fCjjrgLkYsZo22ciMjE9VNg4z9
1x+aja5E7olxX6mLq61JvXLA+fLUrLYOa5kUs3+vcSjr1FkEGKOB8jYUcsnwx1Ot
jvSFZpV0
=Ghuq
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `3fb7d2e970f192a638e72cead0f90dc92194e177f7fcf6b6a23b27665d46f3a5  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401083fb7d2e970f192a638e72cead0f90dc92194e177f7fcf6b6a23b27665d46f3a5f0106cb4a986c14a2834666ea01bd404f71e08fff0100299478a72d5ef7eab0912191dfc2fb508f1045f548e93f0088d2630a2e6a383b00083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0100ac7c9dd79860b768ebfa2821ad0594808f1045f548e92f008bd16d2771ab072f30083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff01026e6b7b0aab240e721bfbd71b6b853f908f1045f548e93f008de7c0a6e4ecdb3eb0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df01036b7f6e0865222e621936e37ba1ac2c808f1045f548e93f0085f835eeeceb0c5600083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

**Comment by jnewbery** [comment]
rebased on master.

Also updated `NodeContext.peer_logic` to `NodeContext.peerman` to be consistent with the new class name.

**Comment by MarcoFalke** [comment]
re-ACK bb6a32ce99 only change is rebase due to conflict in struct NodeContext and variable rename 🤸

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK bb6a32ce99 only change is rebase due to conflict in struct NodeContext and variable rename 🤸
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUj5EwwAgJp5hQNNMrXFOSwWHrMZfCS1OfNXgNYhlArGTaksEIeWnJDhRQ4IsH+V
Y/9eTNx/HALYtCI6B+JcJ9BKpKFyvkCcBAReHCvpq0f47Q97bBcfn6630pH03c++
joxTTZxFv8shXm5VTvr+5GFDJFQZ6UvnoPqByZHvLvB69knjpHqX6E46m2Ki5QYE
pK5FKoD2wFMjj82+YDoELfVx98r6zw0h7uVeFQbBH4I0EL28RYsXTj5WAAF+wWCj
Nzfx+ELzNtjpPQ4Kp4PItWSpJWa96+7HD/UmZB6dWcqE38sLF+nMWFHR1r1oDRCW
s45QqvuQUYDgQW42OnxNd4O8t0vMO6yc0VM5VWd7w7jEDzh03MQSBzrhOIr9Cnd9
QolSfQmFuvSHnXCoyM8NHC4CW6FFarQU0mjmb765p1sX5Z6im01ivS15c1KCKuCA
SmsoydGdO7GtKV6ZdG8ut+ztkvBK1146HxpxZBmDjIC9P0nnHzzYIjAztq/qd4sv
iU/NLnck
=HoQu
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `65490dd75525720cc364871d69f74f20cb23d460478111dc409134823950f8b3  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e89294010865490dd75525720cc364871d69f74f20cb23d460478111dc409134823950f8b3f0107047377b7290bfedd1b3cdee2d3437ff08fff0101955fa1f91352e6dcb81c724bdcb1d7708f1045f565057f008f83d1b213c4b03b10083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff01042a9d680361de9d527650e8f001d7bb408f1045f565056f0083bde561f970f75510083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010650ac79ac5501426fc00857d1c9644d008f1045f565058f008c6d4f49210d647aa0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010f7130b1d65682efe601366e6525b848408f1045f565058f00802a6f5aa1117554e0083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Review by hebasto — APPROVED** [review]
re-ACK bb6a32ce9983c72afa90f41a43a47ffd703ca006, only rebased, and added renaming `s/peer_logic/peerman/` into scripted-diff since my [previous](https://github.com/bitcoin/bitcoin/pull/19791#pullrequestreview-483118079) review (verified with `git range-diff`).

**Pull request merged**